src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/LabelTests.cs

```diff
@@ -283,6 +283,40 @@ public void Label_ImageAlign_Set_GetReturnsExpected(bool autoSize, ContentAlignm
             Assert.False(control.IsHandleCreated);
         }
 
+        [WinFormsFact]
+        public void Label_AutoSize_Does_Not_Change_Size_When_False()
+        {
+            using Form form = new Form();
+            form.Show();
+            using Label label = new();
+            label.Parent = form;
+            label.AutoSize = false;
+            label.Size = new Size(10, 10);
+            label.Text = "Hello";
+            Size oldSize = label.Size;
+            label.Text = "Say Hello";
+            Size newSize = label.Size;
+
+            Assert.Equal(newSize, oldSize);
+        }
+
+        [WinFormsFact]
+        public void Label_AutoSize_Changes_Size_When_True()
+        {
+            using Form form = new Form();
+            form.Show();
+            using Label label = new();
+            label.Parent = form;
+            label.AutoSize = true;
+            label.Size = new Size(10, 10);
+            label.Text = "Hello";
+            Size oldSize = label.Size;
+            label.Text = "Say Hello";
+            Size newSize = label.Size;
+
+            Assert.NotEqual(newSize, oldSize);
+        }
```

Comment:
I think these two can be combined into a theory test, something like
```suggestion
        [WinFormsTheory]
        [InlineData(false, false)]
        [InlineData(true, true)]
        public void Label_AutoSize_BehavesExpected(bool autoSize, bool expected)
        {
            using Form form = new Form();
            using Label label = new()
            {
                AutoSize = autoSize,
                Size = new(10, 10),
                Text = "Hello",
            };
            Size oldSize = label.Size;
            form.Controls.Add(label);
            form.Show();

            label.Text = "Say Hello";
            Size newSize = label.Size;

            Assert.Equal(expected, newSize == oldSize);
        }

```